src/main/scala/org/allenai/pipeline/Pipeline.scala

```diff
@@ -24,23 +27,49 @@ import java.util.Date
 trait Pipeline extends Logging {
   def artifactFactory: FlatArtifactFactory[String] with StructuredArtifactFactory[String]
 
-  def run(rawTitle: String, outputs: Producer[_]*): Unit = {
-    try {
-      val outputInfo = outputs.map(_.stepInfo)
-      require(
-        outputInfo.forall(_.outputLocation.isDefined),
-        s"Running a pipeline without persisting the output: ${
-          outputInfo.filter(_.outputLocation.isEmpty)
-            .map(_.className).mkString(",")
-        }"
-      )
+  private[this] val persistedSteps: ListBuffer[Producer[_]] = ListBuffer()
+
+  /** Run the pipeline.  All steps that have been persisted will be computed, along with any upstream dependencies */
+  def run(title: String): Unit = {
+    run(title, persistedSteps.toSeq)
+  }
 
+  /** Run only specified steps in the pipeline.  Upstream dependencies must exist already.  They will not be computed */
+  def runOnly(title: String, runOnlyTargets: Producer[_]*) {
```

Comment:
Given we can trigger a lot of code from the console, it would be super awesome to establish the full console workflow. What's unclear here:

* How do we set the "tmpOutputDir" so one can quickly iterate over a dataset from the console?
* How do we retrieve the tmp dataset. Ideally, should be as simple as: val matching = pipeline.runOnly("matching")(0)